runtime/lua/vim/lsp/util.lua

```diff
@@ -755,12 +770,23 @@ function M.rename(old_fname, new_fname, opts)
     vim.notify('Rename target already exists. Skipping rename.')
     return
   end
-  local oldbuf = vim.fn.bufadd(old_fname)
-  vim.fn.bufload(oldbuf)
 
-  -- The there may be pending changes in the buffer
-  if vim.fn.isdirectory(old_fname) == 0 then
-    api.nvim_buf_call(oldbuf, function()
+  local oldbufs = {}
+  local win = nil
+
+  if vim.fn.isdirectory(old_fname) == 1 then
+    local cwd = vim.fn.getcwd():gsub('([^%w])', '%%%1')
```

Comment:
Can you add a comment explaining why the cwd/gsub is necessary?

Reply:
Done, hopefully I made it more clear

Reply:
Thanks it does. But I'm not sure if this is safe.

What happens if you start nvim in `/project1/x` and then edit `/project2/y` ? 
And I thought `nvim_buf_get_name` returns the absolute path?

Reply:
I have just checked it again, and you are correct. I have rewrote `get_directory_buffers` according to this